pylint/reporters/ureports/nodes.py

```diff
@@ -14,30 +14,33 @@
 
 A micro report is a tree of layout and content objects.
 """
+from typing import Any, List, Optional, Tuple, Union
+
+from pylint.reporters.ureports.text_writer import TextWriter
 
 
 class VNode:
-    def __init__(self, nid=None):
+    def __init__(self, nid: Optional[Any] = None) -> None:
         self.id = nid
         # navigation
-        self.parent = None
-        self.children = []
-        self.visitor_name = self.__class__.__name__.lower()
+        self.parent: Optional[Any] = None
+        self.children: List = []
+        self.visitor_name: str = self.__class__.__name__.lower()
 
     def __iter__(self):
         return iter(self.children)
 
-    def append(self, child):
+    def append(self, child: Any) -> None:
         """add a node to children"""
         self.children.append(child)
         child.parent = self
 
-    def insert(self, index, child):
+    def insert(self, index: int, child: Union["Paragraph", "Title"]) -> None:
         """insert a child node"""
         self.children.insert(index, child)
         child.parent = self
 
-    def accept(self, visitor, *args, **kwargs):
+    def accept(self, visitor: TextWriter, *args: Any, **kwargs: Any) -> Optional[Any]:
```

Comment:
```suggestion
    def accept(self, visitor: TextWriter, *args: Any, **kwargs: Any) -> None:
```
Do `visit_...` functions ever return anything? I don't believe so right?
Same goes for `leave()` just below here.

Reply:
I used pyannotate so it probably return something once. Let's find out where and fix it :)